collect_app/src/main/java/org/odk/collect/android/widgets/utilities/ExternalAppRecordingRequester.java

```diff
@@ -50,4 +52,14 @@ public void denied() {
             }
         });
     }
+
+    @Override
+    public void onIsRecordingChanged(Consumer<Boolean> isRecordingListener) {
+        isRecordingListener.accept(false);
+    }
+
+    @Override
+    public void onRecordingAvailable(FormEntryPrompt prompt, Consumer<String> recordingAvailableListener) {
+        // This could be implemented using the new Activity Result API  once it's stable
```

Comment:
This basically. We could massively simplify `AudioWidget`'s implementation by just speaking to this interface.